src/components/actionSheet/index.tsx

```diff
@@ -239,6 +245,41 @@ class ActionSheet extends Component<ActionSheetProps> {
       </Dialog>
     );
   }
+
+  renderNewDialog() {
+    const {visible, onDismiss, dialogStyle, onModalDismissed, testID, useSafeArea, dialogProps} = this.props;
+
+    if (onModalDismissed) {
+      LogService.deprecationWarn({component: 'ActionSheet', oldProp: 'onModalDismissed', newProp: 'onDismiss'});
+    }
```

Comment:
`onModalDismissed` comes from the `ActionSheet`, it is no longer needed with the new `Dialog`.
We cannot omit it because that will cause a TS error, I'll add `deprecated` to it.

Reply:
I understand, but if a user uses our `ActionSheet` and passes those both props, with a different callback for each (for some reason) he won't be able to do so anymore.
I agree it's a bit rare and not something that we should care about much, but at least mention somewhere that it's a breaking change.. don't you think?

Reply:
But they user does not have to send the `migrateDialog` prop, this will only be "mandatory" in v7, where we can break the user.
Not sure if this is clear, but this is not our private `ActionSheet`.